spark/src/main/scala/io/delta/tables/execution/VacuumTableCommand.scala

```diff
@@ -32,28 +32,14 @@ import org.apache.spark.sql.types.StringType
  * }}}
  */
 case class VacuumTableCommand(
-    path: Option[String],
-    table: Option[TableIdentifier],
+    pathToVacuum: Path,
```

Comment:
The child logical plan is optional: 
```
    : VACUUM (path=STRING | table=qualifiedName)
        (RETAIN number HOURS)? (DRY RUN)?                               #vacuumTable
```
When there is path, the child plan would be come None. We don't really support optional children.

Reply:
As I took a look at the implementation of VacuumTableCommand, it only cares about the `pathToVacuum`.
I didn't use `UnresolvedDeltaPathOrIdentifier` since I introduced `UnresolvedDeltaIdentifier` in this PR.